src/rest-list.c

```diff
@@ -112,3 +112,26 @@ void rest_list_remove(rest_list_t *list, void *data)
     assert(false);
 }
 
+void rest_list_append(rest_list_t *list, rest_list_t *extension)
```

Comment:
Is this function used anywhere? 
Intuitively append operation is not destructive, but in this implementation `extension` list is destroyed. IMO it could be more similar to std::list [merge](https://en.cppreference.com/w/cpp/container/list/merge) where elements from one list are added to another or like Node.js [Buffer.concat](https://nodejs.org/api/buffer.html#buffer_class_method_buffer_concat_list_totallength) where a third buffer is created by copying contents of the two input buffers. Both of them are non-destructive.

Reply:
It is not used, but I decided to leave it in case we will need it. I can rewrite so it is more intuitive.

Reply:
If it's not used, it's better to remove it. Less commit noise and untested code. I think it's worth referencing this function in #37 as additional linked-list API function.

Reply:
Ok, will remove